draft/XXXX-configuration_handling/README.md

```diff
@@ -0,0 +1,202 @@
+---
+RFC: XXXX
+Author: Roni Choudhury
+Created: 2020-06-29
+Created: 2020-12-09
+Last-Modified: 2020-12-09
+---
+
+# RFC XXXX: Configuration Handling
+
+One of the core features of our vision for Tinker Engine is the ability to
+accept a kind of "meta-configuration" from which concrete configurations are
+generated, used to drive individual protocol runs, with results aggregated
+together in some way that enables post-hoc data analysis. This RFC lays out some
+ideas for some of the core features such a handler will have, along with an
+overall architecture for configuration hanlding in Tinker Engine.
+
+## Proposals
+
+This RFC will discuss a few concrete proposals. One is the overall architcture
+by which the Tinker Engine runtime receives, processes, and submits
+configuration data to a protocol. Another is the method by which a configuration
+file can express variation in its parameters. A third is a fallback mechanism
+allowing programmatic configuration handling when the declarative methods are
+not sufficient for some use case.
+
+### Overall Architecture
+
+Currently, invoking the `tinker` CLI requires specifying a configuration file,
+but currently Tinker does nothing with this file. The new behavior will be for
+Tinker to load the specified file and parse it as YAML (bearing in mind that
+valid JSON is also valid YAML). If the file cannot be opened, or if the contents
+do not parse properly, Tinker will exit with an error condition.
+
+The YAML content describes a "meta-configuration" (see below) that requires
+processing to turn it into a sequence of configurations, one per experimental
+run. This sequence of configurations will be used to construct one protocol per
+configuration, running each to complete one trial of the experimental protocol.
+
+This will be accomplished by sending each configuration dictionary to the
```

Comment:
Multi-level dictionary*

Reply:
Right, this is implied in what I wrote. In general, the config will describe a "tree" of parameters.

Reply:
Ok, that's fine then.